### ISSUE TITLE
Produce list of requests with no response after 25 working days with most recent first

### ISSUE THREAD

**Issue by RichardTaylor** [mysociety/alaveteli#3479]
This is being requested to help administrators find public bodies with non-working email addresses. 

Some public body email addresses become disused and no error / autoresponse message is prompted by messages to them. Sometimes the email address in the system is wrong, but sending messages to it produces no error. 

Often users do report a problem to the administration team, but sometimes users' don't and we only find out about problems many months after it would have been good to have been prompted to fix the contact address. 

This ticket arises from a WhatDoTheyKnow catch up call.

The solution could be a one-off database extract, an admin page, or a public page. 

I will raise a separate ticket for encouraging users to get in touch if they've not had any response at all.

The user experience label is added as it's a poor user experience when your request gets sent to a non-functional email address. 

WhatDoTheyKnow has a specific problem with mail sent to disused/wrong NHS email addresses not prompting an error, or any other, response.

**Comment by lizconlan** [comment]
I've hacked together a basic script for extracting the data on the console. Limited to 100 items while still working out the best/most useful approach.

```rb
results = InfoRequest
            .where("info_requests.prominence = 'normal' AND " \
                   "last_public_response_at is NULL AND " \
                   "created_at < ? ", Date.today - 25.days)
            .order("created_at DESC").limit(100)

site = 'https://www.whatdotheyknow.com/request/'
rows = results.map { |result| [result.id, site + result.url_title, result.created_at] }

CSV.open('recent_unanswered.csv', 'w', :force_quotes => true) do |csv|
  csv << ["id", "url", "created"]
  rows.each { |row| csv << row }
end
```

**Comment by lizconlan** [comment]
Updated script:

```rb
results = InfoRequest.where("info_requests.prominence = 'normal' AND " \
                   "last_public_response_at is NULL AND " \
                   "described_state = 'waiting_response' AND " \
                   "public_body_id NOT IN (1, 2, 55523) AND " \
                   "created_at < ? ", Date.today - 30.days).order("created_at DESC").limit(100)

root = 'https://www.whatdotheyknow.com/request/'
rows = results.map do |result|
  [
    result.id,
    root + result.url_title,
    result.public_body.name,
    result.user.url_name,
    result.created_at
  ]
end

CSV.open('recent_unanswered.csv', 'w', :force_quotes => true) do |csv|
  csv << ["id", "url", "public body name", "user", "created"]
  rows.each { |row| csv << row }
end
```

(1, 2, 55523 are known to be test authorities so reporting on their requests isn't helpful)

**Comment by lizconlan** [comment]
Closing this for now as we have a reasonable data extract. Can be reopened if there are problems with it

**Comment by garethrees** [comment]
Conversation and links in https://groups.google.com/a/mysociety.org/forum/#!topic/alaveteli/KbskDXztyrY

**Comment by RichardTaylor** [comment]
This list of requests was discussed at the WDTK catch up call. 

Summary - this approach didn’t really help us identify bodies with broken contact details (other things like user error reports and our alerts for terms in error/bounce messages are better for this, but don’t catch silent failures). It did though throw up lots of other problems, mainly requests which aren’t suited for our site, or which public bodies decided to ignore for some reason eg. vexatious, personal correspondence. 

We discussed if we wanted an admin page listing these requests, or for admins to be alerted to requests with no responses after X days.  We were concerned about moving towards proactive moderation of the site (something which is problematic legally) 

We concluded it might be useful to ask for this extract again in say a year’s time to see if any more issues come up. 

The main suggestion arising from the work it is that at #3542  - adding a "report this request" link to outgoing emails - giving public bodies an option other than just not responding when faced with inappropriate correspondence.
